frontends/web/src/components/password.jsx

```diff
@@ -309,3 +304,8 @@ function MatchesPattern({ regex, value = '', text }) {
         <p style="color: var(--color-error);">{text}</p>
     );
 }
+
+function hasCaps(event) {
+    const char = String.fromCharCode(event.which);
```

Comment:
Isn't `.which` is a deprecated property and you should use something else like `.code`?
See https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which and https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code